lib/providers/gcm.js

```diff
@@ -70,6 +70,7 @@ GcmProvider.prototype._createMessage = function(notification) {
   Object.keys(notification).forEach(function (key) {
     if (notification[key] !== null && typeof notification[key] !== 'undefined') {
       message.addData(key, notification[key]);
+      message.addNotification(key, notification[key]);
```

Comment:
I think it's good enough as it is.
No need to separate the notification payload for android and ios.